frontend/projects/upgrade/src/app/core/experiments/experiment-utility.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { ExperimentPartition, ExperimentCondition } from './store/experiments.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ExperimentUtilityService {
```

Comment:
I created an additional service to off-load some of the business logic that is clogging up the component classes that doesn't really need to be there. This is especially true for validation logic and other functions that are not directly related to view-model logic.